pkg/skaffold/runner/dev.go

```diff
@@ -109,6 +109,9 @@ func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer, logger *kuber
 			event.DevLoopFailedInPhase(r.devIteration, sErrors.Build, err)
 			return nil
 		}
+		r.changeSet.needsRedeploy = true
+		needsDeploy = deployIntent && r.changeSet.needsRedeploy
```

Comment:
Since we just set `r.changeSet.needsRedeploy = true`, can this be simplified to:
```suggestion
		needsDeploy = deployIntent
```